text/0000-rustdoc-json.md

```diff
@@ -0,0 +1,1124 @@
+`name`    | String  | The name of the Item, if present. Some Items, like impl blocks, do not have names.
+`span`    | [Span](#Span) | (*Optional*) The source location of this Item.
+`visibility` | String | `"default"`, `"public"`, `"crate"`, or `"restricted"` (`pub(path)`).
+`restricted_path` | String | (*Optional*) If `visitility == "restricted"`, this field contains the path that it's restricted to.
+`docs`    | String  | The extracted documentation text from the Item.
+`attrs`   | [String] | The attributes (other than doc comments) on the Item, rendered as strings.
+`deprecation` | [Deprecation](#Deprecation) | (*Optional*) Information about the Item's deprecation, if present.
+`kind`    | String  | The kind of Item this is. Determines what fields are in `inner`.
+`inner`   | Object  | The type-specific fields describing this Item. Check the `kind` field to determine what's available.
+
+### `kind == "module"`
+
+Name     | Type   | Description
+---------|--------|--------------------------------------------------------------------------------
+`items`  | [[ID](#ID)] | The list of Items contained within this module. The order of definitions is preserved.
+
+### `kind == "function"`
+
+Name       | Type     | Description
+-----------|----------|----------------------------------------------------------------------------
+`decl`     | [FnDecl](#FnDecl) | Information about the function signature, or declaration.
+`generics` | [Generics](#Generics) | Information about the function's type parameters and `where` clauses.
+`header`   | String   | `"const"`, `"async"`, `"unsafe"`, or a space separated combination of those modifiers.
+`abi`      | String   | The ABI string on the function. Non-`extern` functions have a `"Rust"` ABI, whereas `extern` functions without an explicit ABI are `"C"`. See [the reference](https://doc.rust-lang.org/reference/items/external-blocks.html#abi) for more details.
+
+### `kind == "struct" || "union"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`struct_type` | String   | Either `"plain"` for braced structs, `"tuple"` for tuple structs, or `"unit"` for unit structs.
+`generics`    | [Generics](#Generics) | Information about the struct's type parameters and `where` clauses.
+`fields_stripped` | bool | Whether any fields have been removed from the result, due to being private or hidden.
+`fields`      | [[ID](#ID)] | The list of fields in the struct. All of the corresponding Items have `kind == "structfield"`.
+`impls`       | [[ID](#ID)] | All impls (both trait and inherent) for this type. All of the corresponding Items have `kind = "impl"`
+
+### `kind == "structfield"`
+
+Name          | Type     | Description
+--------------|----------|-------------------------------------------------------------------------
+`type`        | [Type](#Type) | The type of this field.
```

Comment:
~~Shouldn't this also have the name of the field?~~ Never mind, the name is on the `Item`.